lib/set.js

```diff
@@ -59,13 +67,12 @@ Set.prototype.addMigration = function(title, up, down){
  */
 
 Set.prototype.save = function(fn){
-  var self = this
-    , json = JSON.stringify(this);
-  fs.writeFile(this.path, json, function(err){
-    if (err) return fn(err);
-
+  var self = this;
+  this.state.save({
```

Comment:
@crunchable Yes that is indeed a bug, or at least poor behaviour. I would really like it to save which migrations have been applied instead of just saving an arbitrary position.

This change should really happen before we merge custom state persistence since that will probably break all custom implementations.